trin-core/src/types/header.rs

```diff
@@ -28,44 +29,76 @@ pub struct Header {
     /// Block receipts root.
     pub receipts_root: H256,
     /// Block bloom filter.
-    pub log_bloom: Bloom,
+    pub logs_bloom: Bloom,
     /// Block difficulty.
     pub difficulty: U256,
     /// Block number.
+    #[serde(deserialize_with = "de_hex_to_u64")]
     pub number: u64,
```

Comment:
We can get rid of `de_hex_to_u64` if we change this type to `ethereum_types::U64`.

Reply:
Hmm, why do we deserialize from hex to u64 but keep serializing u64 to u64? I think it is better if we are consistent here and use [serde_hex](https://docs.rs/serde-hex/latest/serde_hex/) to ser/de from u64 to hex and the reverse instead of writing our own implementations...

Reply:
Well, on one hand we only serialize this to display the header to a user. In which case, I'm ok with showing the int and not the hex value, it's way easier than having to go convert the hex value to figure out the block number. Even web3.py does some formatting to hex values when they're returned to the user.

I also think it's preferable to avoid importing a whole library if we can replace it with a single method...